shared/devices/render.native.js

```diff
@@ -86,83 +45,112 @@ const RevokedDescription = () => (
   </Box>
 )
 
-const RevokedDevices = ({revokedDevices}) => (
-  <RevokedHeader>
-    <RevokedDescription />
-    {revokedDevices.map(device => <DeviceRow key={device.name} device={device} revoked={true} />)}
-  </RevokedHeader>
-)
+type RevokedHeaderState = {expanded: boolean}
+class RevokedDevices extends Component<void, {revokedDevices: Array<Object>}, RevokedHeaderState> {
+  state: RevokedHeaderState;
+
+  constructor (props: Props) {
+    super(props)
+    this.state = {expanded: false}
+  }
 
-const DeviceHeader = ({addNewDevice}) => (
-  <Box style={stylesCommonRow}>
+  _toggleHeader (e) {
+    this.setState({expanded: !this.state.expanded})
+  }
+
+  render () {
+    if (!this.props.revokedDevices) {
+      return null
+    }
+
+    const iconType = this.state.expanded ? 'iconfont-caret-down' : 'iconfont-caret-right'
+
+    return (
+      <Box>
+        <TouchableHighlight onPress={e => this._toggleHeader(e)}>
+          <Box style={stylesRevokedRow}>
+            <Text type='BodySemibold'>Revoked devices</Text>
+            <Icon type={iconType} style={{padding: 5}} />
+          </Box>
+        </TouchableHighlight>
+        {this.state.expanded && <RevokedDescription />}
+        {this.state.expanded && this.props.revokedDevices.map(device => <DeviceRow key={device.name} device={device} revoked={true} />)}
+      </Box>)
+  }
+}
+
+const DeviceHeader = ({addNewPhone, addNewComputer, addNewPaperKey}) => {
+  const items = [
+    {title: 'New Phone', onClick: () => addNewPhone()},
+    {title: 'New Computer', onClick: () => addNewComputer()},
+    {title: 'New Paper Key', onClick: () => addNewPaperKey()},
+    {title: 'Cancel', onClick: () => {}},
+  ]
+
+  return <Box style={{...stylesCommonRow, alignItems: 'center'}}>
     <Box style={stylesCommonColumn}>
       <Icon type='icon-devices-add-64-x-48' />
     </Box>
     <Box style={stylesCommonColumn}>
-      <Text type='BodyPrimaryLink' onPress={addNewDevice}>Add new...</Text>
+      <Text type='BodyPrimaryLink' onClick={() => ActionSheetIOS.showActionSheetWithOptions({
```

Comment:
Hm, here's an iOS import in a shared render.native.js -- what does it do on Android?